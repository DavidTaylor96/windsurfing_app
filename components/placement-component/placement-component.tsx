import React, { FC } from 'react';
import { StyleSheet } from 'react-native';
import { View } from '../../components/Themed';

export const Row: FC = (props) => {
  return <View style={styles.row}>{props.children}</View>;
};

export const Column: FC = (props) => {
  return <View style={styles.column}>{props.children}</View>;
};

export const Flex = () => {
  return <View style={{ flex: 1 }} />;
};

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    flex: 1,
  },

  column: {
    flexDirection: 'column',
    justifyContent: 'center',
    flex: 1,
  },
});
