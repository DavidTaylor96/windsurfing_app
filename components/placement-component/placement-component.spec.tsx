import * as React from 'react';
import renderer from 'react-test-renderer';
import { Column, Flex, Row } from './placement-component';

it(`Row`, () => {
  const row = renderer.create(<Row />);

  expect(row).toMatchSnapshot();
});
it(`Column`, () => {
  const column = renderer.create(<Column />);

  expect(column).toMatchSnapshot();
});
it(`Flex`, () => {
  const flex = renderer.create(<Flex />);

  expect(flex).toMatchSnapshot();
});
