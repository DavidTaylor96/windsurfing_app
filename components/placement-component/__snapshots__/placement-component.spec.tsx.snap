// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
/>
`;

exports[`Flex 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;

exports[`Row 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
/>
`;
